pkg/virt-controller/watch/migration.go

```diff
@@ -1687,3 +1761,28 @@ func (c *MigrationController) isMigrationPolicyMatched(vmi *virtv1.VirtualMachin
 	migrationPolicyName := vmi.Status.MigrationState.MigrationPolicyName
 	return migrationPolicyName != nil && *migrationPolicyName != ""
 }
+
+func (c *MigrationController) isMigrationAbortedBeforeHandoff(migrationKey string) bool {
+	// Read operations on maps are thread-safe. For more info: https://go.dev/doc/faq#atomic_maps
+
+	_, isAborted := c.abortedMigrationsBeforeHandoff[migrationKey]
+	return isAborted
```

Comment:
I'm unsure if this is safe. 

```. As long as all goroutines are only reading—looking up elements in the map, including iterating through it using a for range loop—and not changing the map by assigning to elements or doing deletions, it is safe for them to access the map concurrently without synchronization.```

there are goroutines mutating the map while other goroutines are reading the map. I believe that's why we need a lock during reads as well.

Reply:
:+1: